src/main/java/frc/robot/subsystems/HandlerSubsystem.java

```diff
@@ -13,84 +16,166 @@
-
     m_encoder = m_motor.getEncoder();
 
-    //TODO: sensor configs here
+    m_algaeProxSensor = new Pololu4079(
+      HandlerConstants.kAlgaeSensorInput, HandlerConstants.kAlgaeSensorProxThreshold, HandlerConstants.kDebounceTime);
+    m_backProxSensor = new Pololu4079(
+      HandlerConstants.kBackSensorInput, HandlerConstants.kBackSensorProxThreshold, HandlerConstants.kDebounceTime);
+    m_frontProxSensor = new Pololu4079(
+      HandlerConstants.kFrontSensorInput, HandlerConstants.kFrontSensorProxThreshold, HandlerConstants.kDebounceTime);
+    m_distanceSensor = new Pololu4079(
+      HandlerConstants.kDistanceSensorInput, HandlerConstants.kDistanceSensorProxThreshold, HandlerConstants.kDebounceTime);
   }
 
   public void initializePreloaded() {
     m_state = m_chooser.getSelected();
   }
 
-  public void intake() {
+  public double getDistanceSensorMeasurement() {
+    return m_distanceSensor.getDistance();
+  }
+
+  public Angle getHandlerPosition() {
+    return Angle.ofBaseUnits(m_encoder.getPosition(), Units.Radians);
+  }
+
+  public AngularVelocity getHandlerVelocity() {
+    return AngularVelocity.ofBaseUnits(m_encoder.getVelocity(), Units.RadiansPerSecond);
+  }
+
+  public void intakeCoral() {
+    switch (m_state) {
+      case EMPTY:
+      case CANCELLING_CORAL: {
+        m_motor.set(m_intakeSpeedCoral);
+        m_state = State.INTAKING_CORAL;
+        break;
+      }
+      case INTAKING_CORAL:
+      case INTAKING_ALGAE:
+      case LOADING_CORAL:
+      case LOADED_CORAL:
+      case LOADED_ALGAE:
+      case EJECTING_CORAL:
+      case EJECTING_ALGAE:
+      case CANCELLING_ALGAE: {
+        break;
+      }
+    }
+  }
+
+  public void intakeAlgae() {
     switch (m_state) {
       case EMPTY:
-      case CANCELLING: {
-        m_motor.set(m_intakeSpeed);
-        m_state = State.INTAKING;
+      case CANCELLING_ALGAE: {
+        m_motor.set(m_intakeSpeedAlgae);
+        m_state = State.INTAKING_ALGAE;
         break;
       }
-      case INTAKING:
-      case LOADING:
-      case LOADED:
-      case EJECTING: {
+      case INTAKING_CORAL:
+      case INTAKING_ALGAE:
+      case LOADING_CORAL:
+      case LOADED_CORAL:
+      case LOADED_ALGAE:
+      case EJECTING_CORAL:
+      case EJECTING_ALGAE:
+      case CANCELLING_CORAL: {
         break;
       }
     }
   }
 
   public void cancelIntake() {
     switch (m_state) {
-      case INTAKING: {
+      case INTAKING_CORAL:
+      case INTAKING_ALGAE: {
         m_motor.stopMotor();
-        m_state = State.CANCELLING;
+        m_state = State.CANCELLING_CORAL;
         break;
       }
       case EMPTY:
-      case LOADING:
-      case LOADED:
-      case EJECTING:
-      case CANCELLING: {
+      case LOADING_CORAL:
+      case LOADED_CORAL:
+      case LOADED_ALGAE:
+      case EJECTING_CORAL:
+      case EJECTING_ALGAE:
+      case CANCELLING_CORAL:
+      case CANCELLING_ALGAE: {
         break;
       }
     }
   }
 
-  public void eject() {
```

Comment:
I wonder if we should just have one `eject()` method that handles both coral and algae ejection. Making them separate just seems to me like it makes using the API harder.